src/jshint.js

```diff
@@ -3528,19 +3528,20 @@ var JSHINT = (function() {
     var inexport = context && context.inexport;
     var isLet = type === "let";
     var isConst = type === "const";
-    var tokens, lone, value, letblock;
+    var isLetExpr = context && context.letExpr;
+    var isLetBlock = isLet && checkPunctuator(state.tokens.next, "(");
+    var tokens, lone, value;
 
-    if (!state.inES6()) {
+    if (!isLetExpr && !isLetBlock && !state.inES6()) {
       warning("W104", state.tokens.curr, type, "6");
     }
 
-    if (isLet && state.tokens.next.value === "(") {
```

Comment:
I'd rather we just got rid of the parsing code for handling this, rather than adding a new setting to allow it. The reason is, we really want to discourage extensions and other users from making use of this at all. There isn't really a good reason to now, even for ESR releases of Firefox